core/src/main/java/io/questdb/griffin/engine/groupby/FillRangeRecordCursorFactory.java

```diff
@@ -93,7 +94,34 @@ public FillRangeRecordCursorFactory(
         this.timestampIndex = timestampIndex;
         this.valueFuncs = fillValues;
         this.metadata = metadata;
+
+        // only do this for value fill
+        if (!(valueFuncs.size() == 1 && valueFuncs.get(0).isNullConstant())) {
+            if (metadata.getColumnCount() - 1 > valueFuncs.size()) {
+                throw SqlException.$(0, "not enough fill values");
+            }
+
+            int passedTimestamp = 0;
```

Comment:
not enough tests for the `passedTimestamp` calculations. Also, please comment what the code on what the logic for `passedTimestamp`